plugins/inputs/github/github.go

```diff
@@ -191,6 +218,33 @@ func getFields(repositoryInfo *github.Repository) map[string]interface{} {
 	}
 }
 
+func (g *GitHub) getPullRequestFields(ctx context.Context, owner, repo string) (map[string]interface{}, error) {
+	options := github.SearchOptions{
+		TextMatch: false,
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+			Page:    1,
+		},
+	}
+
+	classes := []string{"open", "closed"}
+	fields := make(map[string]interface{})
+	for _, class := range classes {
+		q := fmt.Sprintf("repo:%s/%s is:pr is:%s", owner, repo, class)
+		searchResult, response, err := g.githubClient.Search.Issues(ctx, q, &options)
+		if err != nil {
```

Comment:
Looks like we need to check that we hit the rate limit here too?

```suggestion
		if _, ok := err.(*github.RateLimitError); ok {
			g.RateLimitErrors.Incr(1)
		}
		if err != nil {
```

Reply:
Seem like I messed-up the commit and lost `handleRateLimit()` on the way. Brought it back to fix this instance.